xl-objects.go

```diff
@@ -380,6 +415,10 @@ func (xl xlObjects) deleteObject(bucket, object string) error {
 	if ok, err := isMultipartObject(xl.storage, bucket, object); err != nil {
 		return err
 	} else if !ok {
+		metaJSONFile := path.Join(object, "meta.json")
+		if err = xl.storage.DeleteFile(bucket, metaJSONFile); err != nil {
+			return err
+		}
 		if err = xl.storage.DeleteFile(bucket, object); err != nil {
 			return err
 		}
```

Comment:
Now this is done, GetObject, GetObjectInfo need to populate these values.  Did you test this with listing files properly?